cmake/external/protobuf.cmake

```diff
@@ -234,7 +234,7 @@ FUNCTION(build_protobuf TARGET_NAME BUILD_FOR_HOST)
     )
 ENDFUNCTION()
 
-SET(PROTOBUF_VERSION 3.1.0)
+SET(PROTOBUF_VERSION 3.5.0)
```

Comment:
need @luotao1 to evaluate the version upgrade here

Reply:
ready in 2e12aebf246b69294ff57ee797105b43ab288b0e